### PR TITLE
feat(policy): introduce expression in additional appeal requirement trigger

### LINKED ISSUE #13: Support expression in additional appeals criteria
**Summary**
Additional appeals are created when the specified criteria are fulfilled. Currently, the criteria only support a few fields around resource. Therefore, it's not possible to create criteria based on other appeal properties such as account_id or account_type.

**Proposed solution**
Introduce `expression` in the `RequirementTrigger` and evaluate using existing expression library. Available parameter would be `$appeal`
Additionally, once we have expression, existing `Conditions` won't be necessary as expression will cover anything support by `Conditions` right now. We can deprecate or even delete it.
```diff
type RequirementTrigger struct {
	ProviderType string
	ProviderURN  string
	ResourceType string
	ResourceURN  string
	Role         string
- 	Conditions   []*Condition
+	Expression   string 
}
```

### PR DESCRIPTION
**Pull request by rahmatrhd**
closes #13 

proton PR: https://github.com/goto/proton/pull/17

### CODE REVIEW

**Comment by mabdh** [comment]
@rahmatrhd I think you forgot to assign reviewers for this

File: domain/policy.go [hunk 1]
```diff
@@ -234,6 +236,22 @@ func (r *RequirementTrigger) IsMatch(a *Appeal) (bool, error) {
 			}
 		}
 	}
+	if r.Expression != "" {
```
Comment (bsushmith) [inline domain/policy.go:239]: What happens if both condition and expression get set in the policy ? Does any one take precedence ?

File: domain/policy_test.go [hunk 2]
```diff
@@ -181,6 +182,103 @@ func TestStep_ResolveApprovers(t *testing.T) {
 	}
 }
 
+func TestRequirementTrigger(t *testing.T) {
```
Comment (bsushmith) [inline domain/policy_test.go:185]: Can you add a test case where condition and expression both are set ?

Comment (rahmatrhd) [reply domain/policy_test.go:185]: updated

File: domain/policy.go (hunk 1 above)
Comment (rahmatrhd) [reply domain/policy.go:239]: each criteria within the RequirementTrigger will be evaluated, and the results combined with `AND`. So if there are more than one criteria provided (e.g. condition and expression) both evaluations have to be `true`

**Review by mabdh — APPROVED** [review]

**Pull request merged**